lib/3scale/backend/worker_async.rb

```diff
@@ -13,17 +12,17 @@ class WorkerAsync
       DEFAULT_MAX_CONCURRENT_JOBS = 20
       private_constant :DEFAULT_MAX_CONCURRENT_JOBS
 
-      RESQUE_REDIS_NAMESPACE = :resque
-      private_constant :RESQUE_REDIS_NAMESPACE
+      DEFAULT_MAX_PENDING_JOBS = 100
+      private_constant :DEFAULT_MAX_PENDING_JOBS
 
       def initialize(options = {})
         trap('TERM') { shutdown }
         trap('INT')  { shutdown }
 
         @one_off = options[:one_off]
-        @jobs = Queue.new # Thread-safe queue
+        @jobs = SizedQueue.new(max_pending_jobs)
 
-        @job_fetcher = options[:job_fetcher] || JobFetcher.new(redis_client: redis_client)
+        @job_fetcher = options[:job_fetcher] || JobFetcher.new
```

Comment:
It will inherit the default redis client like it is done when called by the sync worker. But in this case the inherited client is async.